YARN-3400. [JDK 8] Build Failure due to unreported exceptions in RPCUtil (rkanter)

+  private static <T extends YarnException> T instantiateYarnException(
+      Class<? extends T> cls, RemoteException re) throws RemoteException {
+    return instantiateException(cls, re);
+  }
+
+  private static <T extends IOException> T instantiateIOException(
+      Class<? extends T> cls, RemoteException re) throws RemoteException {
+    return instantiateException(cls, re);
+  }
+
+  private static <T extends RuntimeException> T instantiateRuntimeException(
+      Class<? extends T> cls, RemoteException re) throws RemoteException {
+    return instantiateException(cls, re);
+  }
+
-          throw instantiateException(YarnException.class, re);
+          throw instantiateYarnException(YarnException.class, re);
-          throw instantiateException(
+          throw instantiateYarnException(
-          throw instantiateException(realClass.asSubclass(IOException.class),
+          throw instantiateIOException(realClass.asSubclass(IOException.class),
-          throw instantiateException(
+          throw instantiateRuntimeException(
